tests/unit/lms/services/canvas_test.py

```diff
@@ -13,271 +13,165 @@
 
 
 class TestPublicURLForFile:
-    def test_without_check_in_course_and_without_a_mapped_file_id(
-        self, canvas_service, file_from_current_course, public_url_for_file
-    ):
-        # This is what happens during a normal student assignment launch:
-        # check_in_course is False and there's no mapped_file_id in the DB.
-
-        url = public_url_for_file(file_id=str(file_from_current_course["id"]))
+    # A special file id which causes CanvasAPIPermissionError errors
+    MISSING_FILE_ID = "missing_file"
 
-        # check_in_course was False, so it didn't call list_files() to check
-        # whether the file_id was in the course.
-        canvas_service.api.list_files.assert_not_called()
+    def test_it_calls_the_canvas_api(self, canvas_service, public_url_for_file):
+        url = public_url_for_file(file_id="file_id")
 
-        # It calls the Canvas API with the file_id and returns the public URL.
```

Comment:
I think the fate of this comment is kind the key to the whole business. To some extent this whole test is now this, and therefore it doesn't require much more commentary beyond the name of the test.